### PR TITLE
Allow running pre/post deployment scripts for referenced packages

### LINKED ISSUE #110: Posibility to run post-deployment scripts of dependencies
Can it be possible to run the pre- and post deployment scripts of your dependencies too?

### PR DESCRIPTION
**Pull request by jmezach**
When referencing other database packages, either by using a `ProjectReference` or a `PackageReference`, the pre- and/or post-deployment scripts of those dependencies are not run during deployment. While this is understandable in a deployment pipeline when using `SqlPackage.exe` it can be desirable when developers need to update their local development databases. For example, when the post-deployment script inserts some well-known data into certain tables, a developer might expect to have that data in his local development database. In this PR I've added opt-in support to allow for this.

Fixes #110

### CODE REVIEW

**Comment by jmezach** [comment]
@jeffrosenberg @ErikEJ If you could have a look through the changes that would be great. I'll ask my team to test-drive a pre-release version of this before completing this PR.

File: src/DacpacTool/DacpacTool.csproj [hunk 1]
```diff
@@ -12,23 +12,20 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.SqlServer.DACFX" Version="150.4826.1" />
-    <PackageReference Include="System.CommandLine" Version="2.0.0-beta1.20427.1" />
-    <PackageReference Include="System.Security.Permissions" Version="4.7.0" />
+    <PackageReference Include="Microsoft.SqlServer.DACFX" Version="150.4897.1" />
+    <PackageReference Include="System.CommandLine" Version="2.0.0-beta1.20574.7" />
+    <PackageReference Include="System.Security.Permissions" Version="5.0.0" />
```
Comment (ErikEJ) [inline src/DacpacTool/DacpacTool.csproj:17]: Is this update required?

File: src/DacpacTool/Extensions.cs [hunk 2]
```diff
@@ -27,6 +113,29 @@ public static void AddReference(this TSqlModel model, string referencePath, stri
...
+
+            MethodInfo getMetadataMethod = null;
+            foreach (var reference in references)
+            {
+                if (getMetadataMethod == null)
+                {
+                    getMetadataMethod = reference.GetType().GetMethod("GetMetadata", BindingFlags.Public | BindingFlags.Instance, null, new Type[] { typeof(string) }, null);
+                }
+
+                var fileName = (string)getMetadataMethod.Invoke(reference, new object[] { "FileName" });
```
Comment (ErikEJ) [inline src/DacpacTool/Extensions.cs:132]: Add null check?

**Review by ErikEJ — CHANGES_REQUESTED** [review]

File: src/DacpacTool/DacpacTool.csproj (hunk 1 above)
Comment (jmezach) [reply src/DacpacTool/DacpacTool.csproj:17]: Not necessarily no, but I'd rather stay up to date with our dependencies. Do you have a concern about these updates that I'm not aware of?

File: src/DacpacTool/Extensions.cs (hunk 2 above)
Comment (jmezach) [reply src/DacpacTool/Extensions.cs:132]: Do you mean on the result of `GetMethod()` or on the result if `Invoke()`?

File: src/DacpacTool/DacpacTool.csproj (hunk 1 above)
Comment (ErikEJ) [reply src/DacpacTool/DacpacTool.csproj:17]: I think it can be removed, actually

Comment (jmezach) [reply src/DacpacTool/DacpacTool.csproj:17]: You mean the System.Security.Permissions package?

File: src/DacpacTool/Extensions.cs (hunk 2 above)
Comment (ErikEJ) [reply src/DacpacTool/Extensions.cs:132]: I meant before doing a result.Add()

File: src/DacpacTool/DacpacTool.csproj (hunk 1 above)
Comment (ErikEJ) [reply src/DacpacTool/DacpacTool.csproj:17]: Yes, the DacFX package now brings in the correct, newer version

Comment (jmezach) [reply src/DacpacTool/DacpacTool.csproj]: Yup, seems to work just fine.

File: src/DacpacTool/Extensions.cs (hunk 2 above)
Comment (jmezach) [reply src/DacpacTool/Extensions.cs:132]: Ah, yes, that makes sense. Added that now.

**Review by ErikEJ — APPROVED** [review]

File: src/DacpacTool/PackageBuilder.cs [hunk 3]
```diff
@@ -1,12 +1,11 @@
 ﻿using System;
 using System.IO;
+using System.IO.Packaging;
```
Comment (jeffrosenberg) [inline src/DacpacTool/PackageBuilder.cs:3]: I don't understand this... the commit message says "respond to breaking changes", so is this change related to some upstream dependency?

**Review by jeffrosenberg — APPROVED** [review]
LGTM, just one thing I didn't understand that I would like to get some context for

File: src/DacpacTool/PackageBuilder.cs (hunk 3 above)
Comment (ErikEJ) [reply src/DacpacTool/PackageBuilder.cs:3]: I think the DacFX package no longer includes a copy of the Packaging API

Comment (jmezach) [reply src/DacpacTool/PackageBuilder.cs:3]: Yeah, I think that's what happend as well. After upgrading the DacFX package the  `Microsoft.Data.Tools.Schema.Sql.Packaging` no longer existed. So I figured I'd just replace it with the built-in `System.IO.Packaging` API to see if that works. It turned out to be basically a drop-in replacement. I'm assuming that when they originally built DacFX that the Packaging API didn't exist in .NET yet. But now it does so they decided to replace their own implementation with the one from the framework.

Comment (ErikEJ) [reply src/DacpacTool/PackageBuilder.cs:3]: I had the exact same experience when updating my .dacpac reverse engineering library.

**Comment by jmezach** [comment]
I've added an additional note to the README that `DefaultValue` isn't actually used prior to this version. I though I'd make that clear in the docs. I've heard back from my team that this is working well for our scenario, so as soon as CI completes I'll merge this PR and release this as 1.11.0.
